gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-healthcheck/src/test/java/io/gravitee/gateway/services/healthcheck/http/AbstractManagedEndpointRuleHandlerTest.java

```diff
@@ -220,9 +222,52 @@ void shouldValidateFromRoot(Vertx vertx, VertxTestContext context) throws Throwa
         assertTrue(context.completed());
     }
 
-    private Endpoint createEndpoint(String targetPath) {
-        HttpEndpoint aDefault = new HttpEndpoint("default", "http://localhost:" + wm.getPort() + (targetPath != null ? targetPath : ""));
+    @Test
+    void shouldValidateWithUnderscoreInHostname(Vertx vertx, VertxTestContext context) throws Throwable {
+        // Prepare HTTP endpoint
+        wm.stubFor(get(urlEqualTo("/")).withHost(equalTo("my_local_host")).willReturn(ok()));
+
+        // Prepare
+        EndpointRule rule = createEndpointRule("http://my_local_host", null, true);
+
+        HealthCheckStep step = new HealthCheckStep();
+        HealthCheckRequest request = new HealthCheckRequest("/", HttpMethod.GET);
+
+        step.setRequest(request);
+        HealthCheckResponse response = new HealthCheckResponse();
+        response.setAssertions(Collections.singletonList(HealthCheckResponse.DEFAULT_ASSERTION));
+        step.setResponse(response);
+        when(rule.steps()).thenReturn(Collections.singletonList(step));
+
+        HttpEndpointRuleHandler runner = new HttpEndpointRuleHandler(vertx, rule, templateEngine, environment);
+
+        // Verify
+        runner.setStatusHandler(
+            (Handler<EndpointStatus>) status -> {
+                assertTrue(status.isSuccess());
+                wm.verify(getRequestedFor(urlEqualTo("/")));
+                context.completeNow();
+            }
+        );
+
+        // Run
+        runner.handle(null);
+
+        // Wait until completion
+        assertTrue(context.awaitCompletion(5, TimeUnit.SECONDS));
+        assertTrue(context.completed());
+    }
+
+    private Endpoint createEndpoint(String baseUrl, String targetPath, boolean useSystemProxy) {
+        HttpEndpoint aDefault = new HttpEndpoint("default", baseUrl + (targetPath != null ? targetPath : ""));
         aDefault.setHttpClientOptions(new HttpClientOptions());
+        if (useSystemProxy) {
+            HttpProxy httpProxy = new HttpProxy();
+            httpProxy.setUseSystemProxy(true);
+            httpProxy.setEnabled(true);
+            aDefault.setHttpProxy(httpProxy);
+        }
```

Comment:
Required for wiremock and multi domain proxy
https://wiremock.org/docs/junit-jupiter/#proxy-mode